plugins/inventory/vmware_vm_inventory.py

```diff
@@ -624,6 +645,19 @@ def _get_hostname(self, properties, hostnames):
             if hostname:
                 return to_text(hostname)
 
+    def _can_add_host(self, host_filters, host_properties, host, strict=False):
+        for host_filter in host_filters:
+            try:
+                can_add_host = self._compose(host_filter, host_properties)
+            except Exception as e:  # pylint: disable=broad-except
+                if strict:
+                    raise AnsibleError("Could not evaluate %s as host_filters %s" % (host_filter, to_native(e)))
+                display.warning(to_native(e))
```

Comment:
This could be risky. I need to test it to confirm. There is great hesitancy to do any logging in inventory plugins, because `ansible-inventory` requires valid JSON written to stdout. If stdout and stderr are properly separated, this may work correctly, but I need to run a test case first.

Reply:
still need to test that after other comment about can_add_host not being defined is resolved.

Reply:
okay, with patch for undefined variable, this seems to still not mess up parsing as far as I'm concerned